### PR TITLE
Enable GPTQ in executorch

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
Previously we just added the code but didn't test it, this PR also tests gptq locally to make sure we can produce a model using gptq from torchao repo

Test Plan:
python3 -m examples.models.llama2.export_llama -c stories110M.pt -p params.json -qmode 8da4w-gptq -X -d fp32

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: examples/models/llama2/quantize.py [hunk 1]
```diff
@@ -131,507 +130,6 @@ def dynamically_quantize_per_channel(
...
-    # needed for GPTQ single column quantize
-    if group_size > input.shape[-1] and scales.shape[-1] == 1:
-        group_size = input.shape[-1]
-
-    assert input.shape[-1] % group_size == 0
-    assert input.dim() == 2
-    return torch.empty_like(input, dtype=dtype)
-
-
-def group_quantize_tensor_symmetric(
```
Comment (manuelcandales) [inline examples/models/llama2/quantize.py:509]: deleted here, but still used by prepare_int4_weight_and_scales_and_zeros

File: examples/models/llama2/quantize.py [hunk 2]
```diff
@@ -131,507 +130,6 @@ def dynamically_quantize_per_channel(
...
-    """Get the original weight from the normalized float weight format"""
-    # since we are using int8 we will decode 2 entries per byte
-    # Shift elements down 4 and select out the bottom 4 bits
-    shape = int8_data.shape
-    first_elements = (int8_data >> 4).to(torch.int8)
-    second_elements = (int8_data & 0b1111).to(torch.int8)
-    return torch.stack([first_elements, second_elements], dim=-1).view(up_size(shape))
-
-
-def per_token_dynamic_quant(input: torch.Tensor) -> torch.Tensor:
```
Comment (manuelcandales) [inline examples/models/llama2/quantize.py:616]: deleted here, but still used by linear_forward_8da4w

File: examples/models/llama2/quantize.py [hunk 3]
```diff
@@ -131,507 +130,6 @@ def dynamically_quantize_per_channel(
...
-    scales = max_val_abs / (float(max_int - min_int) / 2)
-    scales = torch.max(scales, torch.full_like(scales, torch.finfo(torch.float32).eps))
-    # TODO: make sure abs(scales) is not too small?
-    zeros = torch.full_like(scales, 0)
-    return scales.to(precision).reshape(w.shape[0], -1), zeros.to(precision).reshape(
-        w.shape[0], -1
-    )
-
-
-def pack_scales_and_zeros(scales, zeros, precision=torch.float16):
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply examples/models/llama2/quantize.py:408]: this is fine, we are not using this QuantHandler, I'll remove later as well

File: examples/models/llama2/quantize.py (hunk 1 above)
Comment (jerryzh168) [reply examples/models/llama2/quantize.py:509]: this is probably not used, we'll cleanup a bit later

File: examples/models/llama2/quantize.py (hunk 2 above)
Comment (jerryzh168) [reply examples/models/llama2/quantize.py:616]: we'll import from torchao

**Review by manuelcandales — APPROVED** [review]

**Review by digantdesai — APPROVED** [review]

File: examples/models/llama2/llama_transformer.py [hunk 4]
```diff
@@ -464,9 +464,9 @@ def forward(
             freqs_cos = self.freqs_cos[sp : sp + seqlen]
             freqs_sin = self.freqs_sin[sp : sp + seqlen]
         else:
-            assert (
-                start_pos is None and cache_k is None and cache_v is None
-            ), "Caches and start_pos are unused when use_kv_cache is False"
+            # assert (
```
Comment (jerryzh168) [inline examples/models/llama2/llama_transformer.py:467]: @kimishpatel is this OK? I need to comment this out to make sure we can run the torch._dyncmo.export in GPTQ.

@HDCharles is going to work on a refactor of GPTQ to remove export and use tensor subclass instead, we can revert this change when that is implemented I think.
